### PR TITLE
Refacto manage version

### PR DESCRIPTION
**Pull request by alavieille**
https://github.com/open-orchestra/open-orchestra/pull/1164
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/749
https://github.com/open-orchestra/open-orchestra-cms-bundle/pull/2160

### CODE REVIEW

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentToolbarView.js [hunk 1]
```diff
@@ -107,14 +101,55 @@ class ContentToolbarView extends OrchestraView
...
+
+    /**
+     * @param {Object} event
+     * @private
+     */
+    _changeStatus(event) {
+        let statusId = $(event.currentTarget).attr('data-id');
+        let status = this._statuses.findWhere({id: statusId});
+        if (typeof status == "undefined") {
+            throw new ApplicationError('Status with id '+statusId+ 'not found');
```
Comment (ngilain) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentToolbarView.js]: @alavieille : statusId+ ' not found'

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js [hunk 2]
```diff
@@ -10,15 +9,19 @@ class ContentVersionsView extends AbstractCollectionView
...
      * @param {string}              contentId
      * @param {string}              language
      * @param {string}              contentTypeId
+     * @param {Object}              siteLanguages
      */
-    initialize({collection, contentId, language, contentTypeId}) {
-        super.initialize({collection: collection});
+    initialize({collection, settings, contentId, language, contentTypeId, siteLanguages}) {
+        super.initialize({collection: collection, settings: settings});
+        console.log(settings);
```
Comment (ngilain) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js]: @alavieille : log to remove

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js [hunk 3]
```diff
@@ -36,18 +39,34 @@ class ContentVersionsView extends AbstractCollectionView
...
-        this._listView = new ContentVersionsListView({
-            collection: this._collection,
-            settings: {
+        let settings = $.extend(true, this._settings, {
                 serverSide: false,
                 processing: false,
                 data: this._collection.models
             }
+        );
+        console.log(settings);
```
Comment (ngilain) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js]: @alavieille : log to remove

File: BackofficeBundle/Resources/public/template/Content/newContentView._tpl.html [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+                        <img src="/images/languages/<%= siteLanguage %>.png" alt="<%= siteLanguage %>" /><%= siteLanguage.toUpperCase() %>
+                    </a>
+                </li>
+            <% } %>
+        <% }); %>
+    </ul>
+</div>
+<h1> <%= Translator.trans('open_orchestra_backoffice.table.contents.new') %></h1>
+<%= renderTemplate('FlashMessage/flashMessage', {flashMessages: messages}) %>
+<div class='form-new'></div>
```
Comment (ngilain) [inline BackofficeBundle/Resources/public/template/Content/newContentView._tpl.html]: @alavieille : NL@EOF

File: BackofficeBundle/Resources/translations/interface.en.yml [hunk 5]
```diff
@@ -232,6 +232,13 @@ open_orchestra_backoffice:
...
+        edit: Edit
         confirm_create:
             title: Create content in a new language
             message: This content does not exist in this language. Would you like to create it?
+        message_online:
+            title: The content is actually online.
+            message: |
+                Editing an online content is not possible. \n
+                To modify this content please create a new version. \n
+                You can also modify the workflow of this version but the content will not be visible on the site
```
Comment (ngilain) [inline BackofficeBundle/Resources/translations/interface.en.yml]: @alavieille : NL@EOF

**Comment by alavieille** [comment]
@ngilain pr updated

File: BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js [hunk 6]
```diff
@@ -36,18 +38,34 @@ class ContentVersionsView extends AbstractCollectionView
      * Render content versions
      */
     render() {
+        let content = this._collection.first();
+        if (typeof content === 'undefined') {
+            throw new ApplicationError('A content should be have at least one version')
```
Comment (ngilain) [inline BackofficeBundle/Resources/public/ecmascript/OpenOrchestra/Application/View/Content/ContentVersionsView.js]: @alavieille : should have

**Comment by ngilain** [comment]
@alavieille : :+1: with the last update

**Pull request merged**